code/processes/gateway.q

```diff
@@ -409,6 +413,12 @@ getserverids:{[att]
   ];
 
   // its a dictionary of attributes
+    // check if attribute types are correct types
+    correctAttTypes:`dates`servertypes`tables!14 11 11h;
+    w:correctAttTypes[key att]=type each att;
+    if[not all{[x;y] y each key x}[att;w];
+    '("Wrong function parameter types provided.", " ",(raze " " sv string (where not w))," parameter(s) need type(s) ",raze " " sv string correctAttTypes where not w),"h"
```

Comment:
Great, run through with some dummy args and that checks out. Last fix for this line, need to fix the indenting and should be able to drop `raze` by moving some brackets around. Can also avoid manually adding `"h"` by using `.Q.s1`.
```suggestion
      '"Wrong function parameter types provided.", " ",(" "sv string where not w)," parameter(s) need type(s) ",.Q.s1 correctAttTypes where not w
```